src/shared-components/dialogModal/test.tsx

```diff
@@ -17,4 +18,15 @@ describe('<DialogModal />', () => {
     getAllByText(modalTitle);
     getAllByText(modalBody);
   });
+
+  it('renders dialog modal with custom color', () => {
+    const { container } = render(
+      <DialogModal modalColor={primaryTheme.COLORS.background}>
+        <div>{modalBody}</div>
+      </DialogModal>,
+    );
+
+    // DOM Traversal necessary to find the React portal element
```

Comment:
Good catch, and hopefully you didn't have to struggle too much to get this working. This actually ties into a WIP feature in the `emotion 10 --> 11` upgrade branch. If you want to simplify it in this PR, feel free to steal the code:

```tsx
// src/tests/testingLibraryHelpers.tsx
import React from 'react';
import { ThemeProvider } from 'emotion-theming';
import * as ReactTestingLibrary from '@testing-library/react';

import { primaryTheme, ThemeType } from '../constants';

interface RenderOptions extends ReactTestingLibrary.RenderOptions {
  theme?: ThemeType;
  withPortalContainer?: boolean;
}

/**
 * Many of our pages rely on components that make use of portals.
 *
 * Unit tests for components that do that **do not include such a portal
 * container in their unit test** will have test failures on CI due to
 * serialization order being off.
 */
const usePortalContainer = () => {
  const portalContainer = document.createElement('div');
  portalContainer.setAttribute('id', 'reactPortalSection');
  document.body.appendChild(portalContainer);

  return portalContainer;
};

// We customize @testing-library methods to bake-in theming and keep unit tests DRY.
// We do not use ReactTestingLibrary.render(Component, { wrapper }) option because
// `@testing-library/react` is (somehow) overwriting React with its own API when used.
// This issue is specific to this repo and is not present in other implementations.
// <ThemeProvider> leaves no DOM trace, anyway, so there's no functional difference.
const customRender = (
  Component: React.ReactElement,
  options: RenderOptions = {},
): ReactTestingLibrary.RenderResult => {
  const {
    theme = primaryTheme,
    container,
    withPortalContainer,
    ...rest
  } = options;

  return ReactTestingLibrary.render(
    <ThemeProvider theme={theme}>{Component}</ThemeProvider>,
    {
      container: withPortalContainer ? usePortalContainer() : container,
      ...rest,
    },
  );
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
```

After adding then, you can pass a `withPortalContainer` boolean prop, and then rely on `container.firstElementChild` as most of our tests do:

```tsx
// src/shared-components/dialogModal/test.tsx
    const { container } = render(
      <DialogModal modalColor={primaryTheme.COLORS.background}>
        <div>{modalBody}</div>
      </DialogModal>,
      { withPortalContainer: true },
    );

    expect(container.firstElementChild).toMatchSnapshot();